python-sdk/src/astro/databases/snowflake.py

```diff
@@ -595,11 +597,20 @@ def load_file_to_table_natively(
         file_path = os.path.basename(source_file.path) or ""
         sql_statement = f"COPY INTO {table_name} FROM @{stage.qualified_name}/{file_path}"
         try:
-            self.hook.run(sql_statement)
+            self.hook.run(sql_statement, handler=self.check_for_error)
         except (ValueError, AttributeError) as exe:
             raise DatabaseCustomError from exe
         self.drop_stage(stage)
 
+    @staticmethod
+    def check_for_error(cur):
+        """
+        Handler to check the error state returned by snowflake when running `copy into` query.
+        """
+        results = cur.fetchall()
+        if any([True for result in results if result["status"] == COPY_INTO_COMMAND_FAIL_STATUS]):
```

Comment:
```suggestion
        if any(result["status"] == COPY_INTO_COMMAND_FAIL_STATUS for result in results):
```

Reply:
Don't use `[...]` in `any` as this would first create a whole list of items before `any` actually runs.

The advantage of `any` is that it allows you to pass an iterable which also can be a generator which means that `any` will report `True` as soon as the first condition is `True`.

Reply:
@feluelle Thanks, didn't know it.